lib/Basic/FileSystem.cpp

```diff
@@ -62,6 +62,19 @@ class LocalFileSystem : public FileSystem {
     }
     return std::unique_ptr<llvm::MemoryBuffer>(result->release());
   }
+
+  virtual bool remove(const std::string& path) override {
```

Comment:
Doesn't this kind of counter the approach of providing the full list of expected files to the stale-file-removal tool? If we recursively delete directories, we might be deleting more files than we want, obviously.